src/main/java/com/salesforce/storm/spout/dynamic/SpoutCoordinator.java

```diff
@@ -158,6 +161,42 @@ public void addVirtualSpout(final DelegateSpout spout) {
     }
 
     /**
+     * Remove a new VirtualSpout from the coordinator. This will signal to the monitor to request that the VirtualSpout
+     * be stopped and ultimately removed.
+     *
+     * This method will blocked until the VirtualSpout has completely stopped.
+     *
+     * @param virtualSpoutIdentifier identifier of the VirtualSpout to be removed.
+     */
+    public void removeVirtualSpout(final VirtualSpoutIdentifier virtualSpoutIdentifier) {
+        if (hasVirtualSpout(virtualSpoutIdentifier)) {
+            throw new SpoutDoesNotExistException(
+                "A spout with id " + virtualSpoutIdentifier + " does not exist in the spout coordinator!",
+                virtualSpoutIdentifier
+            );
+        }
+
+        getSpoutMonitor().removeVirtualSpout(virtualSpoutIdentifier);
+
+        // Let's block until we no longer detect that the spout is in monitor.
+        while (getSpoutMonitor().hasVirtualSpout(virtualSpoutIdentifier)) {
```

Comment:
@Crim is there a better way to do this?

Reply:
So I can think of a couple ways.  You can make the method async and return a future.  Let the caller deal w/ the problem.  Or you can pass in a timeout value for how long to block for and then toss an exception if that's exceeded and let the caller deal w/ it.

Reply:
Not entirely sure how a future would work in this case.  Perhaps you put this while block in the future execution?  Not sure about that one.